src/MouseAdapter.js

```diff
@@ -309,20 +383,196 @@ define(function(require) {
+                if (calculateDeltaDirection(deltaArray[axis],firstRange,5,DELTA_ARRAY_SIZE) === 1) {
+                    // Check the 5 data points prior to those and determine if they were decreasing
+                    var secondRange = getDeltaArrayIndex(-10);
+                    if (calculateDeltaDirection(deltaArray[axis],secondRange,5,DELTA_ARRAY_SIZE) === -1) {
+                        return true;
+                    }
+                }
+                return false;
+            };
+
+            /*
+             * This is a helper function for detecting user scroll events.  When the user
+             * performs a scroll (at least with a track pad), the scroll delta drops to 
+             * nearly 0 before the new momentum is added to the scroll inertia. Because the 
+             * inertia typically decays very slowly over time, a significant decrease in the 
+             * scroll delta is a strong indication of a user event. This function works by 
+             * detecting these negative spikes in the scroll delta.
+             *
+             * One advantage this method has is that it can detect events as soon as they 
+             * occur because it does not take time to collect data.
+             * One disadvantage this method has is that it cannot detect events when the 
+             * the current inertia is below a certain threshold, including really light 
+             * scrolls and scrolls made after the current scroll inertia has decayed a lot.
+             * @param {DELTA_INDEX} axis The axis on which to perform this check
+             * @return {boolean} Returns true if conditions were satisfied to indicate the
+             * user performed a scroll event, else false
+             */
+            var detectNegativeSpike = function(axis) {
+                var lastPoint = getDeltaArrayIndex(-1);
+                // Check if the current event delta is low.  Nearly all of the negative spikes 
+                // seem to reach between 0.5-2.5, so make sure the point is low enough
+                if (deltaArray[axis][lastPoint] <= 5) {
+                    var recentPoint2 = getDeltaArrayIndex(-2);
+                    var recentPoint3 = getDeltaArrayIndex(-3);
+                    // Check to see if the previous two points are significantly higher.
+                    // Checks last TWO points instead of just last one because the deltas
+                    // seem to be prone to intermittent noisy spikes which can create false
+                    // positives.
+                    if ((deltaArray[axis][recentPoint2] / deltaArray[axis][lastPoint] > 2 &&
+                         deltaArray[axis][recentPoint2] - deltaArray[axis][lastPoint] > 2) &&
+                        (deltaArray[axis][recentPoint3] / deltaArray[axis][lastPoint] > 2 &&
+                         deltaArray[axis][recentPoint3] - deltaArray[axis][lastPoint] > 2)) {
```

Comment:
If these were well-named parameters, the logic would be much easier to follow. And during debugging we could see the exact values of each calculation.